clinvar-xml-parser/src/main/java/uk/ac/ebi/eva/clinvar/Application.java

```diff
@@ -0,0 +1,163 @@
+            System.out.println(
+                    "Transforming clinvar file " + inputFileName + " into Json file " + outputFilePath + " ...");
+            multiThreadExecute(reader, clinvarSetTransformer, clinvarJsonSerializer);
+        }
+
+    }
+
+    private void multiThreadExecute(XmlClinVarReader reader, ClinvarSetTransformer clinvarSetTransformer,
+                                    ClinvarJsonSerializer clinvarJsonSerializer) {
+        // submit each task (read, transform, serialize) in a thread
+        executor = Executors.newFixedThreadPool(3);
+        Future<Integer> readRecords = executor.submit(reader);
+        Future<Integer> transformedRecords = executor.submit(clinvarSetTransformer);
+        Future<Integer> writtenRecords = executor.submit(clinvarJsonSerializer);
+
+        try {
+            if (readRecords.get().equals(transformedRecords.get()) &&
+                    readRecords.get().equals(writtenRecords.get())) {
+                System.out.println("Finished successfully: " + readRecords.get() + " clinvar records processed");
+            }
+        } catch (InterruptedException e) {
+            System.out.println("Thread interrupted: " + e.getMessage());
+        } catch (ExecutionException e) {
+            System.out.println("Exception in thread execution: " + e.getMessage());
+        }
+
+        closeExecutor();
+    }
+
+    private ClinvarJsonSerializer getClinvarJsonSerializer(ArrayBlockingQueue<ClinvarSet> clinvarSetsQueue,
+                                                           Path outputFilePath) {
+        // build Json serializer
+        ClinvarJsonSerializer clinvarJsonSerializer;
+        try {
+
+            BufferedWriter bw = new BufferedWriter(
+                    new OutputStreamWriter(new GZIPOutputStream(Files.newOutputStream(outputFilePath))));
+            clinvarJsonSerializer = new ClinvarJsonSerializer(clinvarSetsQueue, bw, this);
+        } catch (IOException e) {
+            System.out.println("Error: output file " + outputFilePath + " cannot be written: " + e.getMessage());
+            return null;
+        }
+        return clinvarJsonSerializer;
+    }
+
+    private ClinvarSetTransformer getClinvarSetTransformer(ArrayBlockingQueue<String> xmlStringsQueue,
+                                                           ArrayBlockingQueue<ClinvarSet> clinvarSetsQueue) {
+        ClinvarSetTransformer clinvarSetTransformer;
+
+        try {
+            // build transformer
+            clinvarSetTransformer = new ClinvarSetTransformer(xmlStringsQueue, clinvarSetsQueue, versionNumber, this);
+        } catch (JAXBException e) {
+            System.out.println("Error: clinvar version " + versionNumber + " cannot be parsed: " + e.getMessage());
+            return null;
+        }
+        return clinvarSetTransformer;
+    }
+
+    private XmlClinVarReader buildXmlClinVarReader(ArrayBlockingQueue<String> xmlStringsQueue) {
```

Comment:
Use the same method naming rules:

* **get**ClinvarJsonSerializer
* **get**ClinvarSetTransformer
* **build**XmlClinVarReader